apps/condo/domains/user/integration/passport/utils/user.js

```diff
@@ -0,0 +1,434 @@
+                await User.update(context, userFoundByPhone.id, {
+                    ...DV_AND_SENDER,
+                    phone: null,
+                    isPhoneVerified: false,
+                    // NOTE: not moved to external, because it might be not external
+                    meta: { ...userFoundByPhone.meta, phone: userFoundByPhone.phone },
+                }, USER_FIELDS)
+                updatePayload.phone = userData.phone
+                updatePayload.isPhoneVerified = providerInfo.trustPhone && userData.isPhoneVerified
+            }
+        } else {
+            updatePayload.phone = userData.phone
+            updatePayload.isPhoneVerified = providerInfo.trustPhone && userData.isPhoneVerified
+        }
+    }
+
+    return await User.update(context, targetUser.id, updatePayload, USER_FIELDS)
+}
+
+/**
+ * Gets or creates a user and their external identity.
+ *
+ * This function first attempts to find a user's external identity.
+ * If found, it returns the associated user.
+ * If not found, it proceeds to find an existing user by their verified and trusted
+ * phone number or email. If no existing user is found, a new user is created.
+ * Finally, it creates the external identity and links it to the user.
+ * @param {import('http').IncomingMessage} req - keystone instance object
+ * @param {object} userProfile - The profile of the user from the external provider.
+ * @param {string} userType - The type of the user (e.g., 'staff', 'resident').
+ * @param {object} providerInfo - information about identity provider (name / trustPhone / trustEmail).
+ * @param {object} fieldMapping - userInfo json remap options. For example { id: 'sub' } converts oidc default "sub" field "id"
+ * @returns {Promise<object>} The resolved user object.
+ */
+async function syncUser (
+    req,
+    userProfile,
+    userType,
+    providerInfo,
+    fieldMapping = {}
+) {
+    const errorContext = { req }
+
+    // Step 0. Sanity checks
+    if (!ALLOWED_USER_TYPES.includes(userType)) {
+        throw new GQLError({
+            ...ERRORS.INVALID_USER_TYPE,
+            messageInterpolation: { allowedTypes: ALLOWED_USER_TYPES.join(', ') },
+        }, errorContext)
+    }
+    const { success: isValidProviderInfo, error: providerInfoError, data: providerInfoData } = providerInfoSchema.safeParse(providerInfo)
+    if (!isValidProviderInfo) {
+        throw new GQLError(ERRORS.INVALID_IDENTITY_PROVIDER_INFO, errorContext, [new Error(z.prettifyError(providerInfoError))])
+    }
+
+    const { success: isValidFieldMapping, error: fieldMappingError, data: fieldMappingData } = fieldMappingSchema.safeParse(fieldMapping)
+    if (!isValidFieldMapping) {
+        throw new GQLError(ERRORS.INVALID_FIELD_MAPPING, errorContext, [new Error(z.prettifyError(fieldMappingError))])
+    }
+
+    // Step 1. Convert provider shape to condo shape
+    const combinedMapping = { ...DEFAULT_USER_FIELDS_MAPPING, ...fieldMappingData }
```

Comment:
IMO just `const mapping` is better here

Reply:
I think current naming is more explicit, and that's the key